pkg/podimpersonation/podimpersonation.go

```diff
@@ -553,27 +562,49 @@ func (s *PodImpersonation) augmentPod(pod *v1.Pod, sa *v1.ServiceAccount, secret
 			},
 		})
 
+	image := imageOverride
+	if image == "" {
+		image = s.imageName()
+	}
+
 	for i, container := range pod.Spec.Containers {
 		for _, envvar := range container.Env {
 			if envvar.Name != "KUBECONFIG" {
 				continue
 			}
 
+			vmount := v1.VolumeMount{
```

Comment:
nit: Could add a comment to explain why we're doing this since it's not obvious at first